integration_test/metadata/integration_metadata.go

```diff
@@ -38,7 +38,7 @@ type ExpectedMetric struct {
 	MonitoredResource string `yaml:"monitored_resource" validate:"required,oneof=gce_instance"`
 	// Mapping of expected label keys to value patterns.
 	// Patterns are RE2 regular expressions.
-	Labels map[string]string `yaml:"labels"`
+	Labels map[string]string `yaml:"labels" validate:"omitempty,gt=0"`
```

Comment:
I think `omitempty` should also be part of the `yaml` annotation (https://pkg.go.dev/gopkg.in/yaml.v2#Marshal)…

Reply:
golang validator also has omitempty. They serve different purpose.

the one yaml uses is useful only during marshalling which we don't have that use case

here is the one for golang validator omitempty: https://pkg.go.dev/github.com/go-playground/validator/v10#hdr-Omit_Empty

Reply:
Here is the context: https://github.com/GoogleCloudPlatform/ops-agent/pull/732#discussion_r929109156

And golang playground:  https://go.dev/play/p/K13XVjHgqkS

![image](https://user-images.githubusercontent.com/8354694/184194111-d811ff69-1ffa-41e8-869d-5c7e63f542de.png)

Reply:
Right, I saw the validator one — that's why I said "also". :grin:

Reply:
Changed in: [16b3b36](https://github.com/GoogleCloudPlatform/ops-agent/pull/760/commits/16b3b36dd2fe40360225c2eeff81a2dde52783ec)